tests/vip-helpers/test-vip-utils.php

```diff
@@ -22,4 +36,170 @@ public function test__valid_attachment() {
 
 		$this->assertEquals( $expected_end_of_url, $actual_end_of_url );
 	}
+
+	/**
+	 * Test vip_safe_wp_remote_request() behavior with normal responses
+	 */
+	public function test__vip_safe_wp_remote_request_with_normal_response() {
+		$url = 'https://localhost';
+		$response = 'mock_response';
+
+		$this->mock_http_response( $response, 0.10 );
+
+		// We can call it 4 times (more than the default threshold of 3) and it always returns the expected response (no failure / fallback)
+		for( $i = 0; $i < 4; $i++ ) {
```

Comment:
Aside: I'm always vary of loops in things like tests (adds too much additional logic), but I think they're okay here as we have things very well-documented; kudos on that btw).